lib/pages/labels/label_widget.dart

```diff
@@ -69,39 +70,62 @@ class LabelExpansionTileWidget extends StatelessWidget {
   }
 }
 
-class LabelRow extends StatelessWidget {
+class LabelRow extends StatefulWidget {
   final Label label;
 
   LabelRow(this.label);
 
+  @override
+  _LabelRowState createState() => _LabelRowState();
+}
+
+class _LabelRowState extends State<LabelRow> {
+  late AppDatabase db;
+
+  var labelDB = LabelDB.get();
+
   @override
   Widget build(BuildContext context) {
     HomeBloc homeBloc = BlocProvider.of(context);
     return ListTile(
-      key: ValueKey("tile_${label.name}_${label.id}"),
+      key: ValueKey("tile_${widget.label.name}_${widget.label.id}"),
       onTap: () {
-        homeBloc.applyFilter("@ ${label.name}", Filter.byLabel(label.name));
+        homeBloc.applyFilter(
+            "@ ${widget.label.name}", Filter.byLabel(widget.label.name));
         Navigator.pop(context);
       },
       leading: Container(
+        child: Icon(
+          Icons.label,
+          size: 20,
+          color: Color(widget.label.colorValue),
+        ),
         width: 24.0,
         height: 24.0,
-        key: ValueKey("space_${label.name}_${label.id}"),
+        key: ValueKey("space_${widget.label.name}_${widget.label.id}"),
       ),
       title: Text(
-        "@ ${label.name}",
-        key: ValueKey("${label.name}_${label.id}"),
+        "@ ${widget.label.name}",
+        key: ValueKey("${widget.label.name}_${widget.label.id}"),
       ),
-      trailing: Container(
-        height: 10.0,
-        width: 10.0,
-        child: Icon(
-          Icons.label,
-          size: 16.0,
-          key: ValueKey("icon_${label.name}_${label.id}"),
-          color: Color(label.colorValue),
+      trailing: IconButton(
+        onPressed: () {
+          Navigator.pop(context);
+          del(widget.label.id);
+        },
+        icon: Icon(
+          Icons.delete,
+          size: 20.0,
+          key: ValueKey("icon_${widget.label.name}_${widget.label.id}"),
+          color: Colors.grey[500],
         ),
       ),
     );
   }
+
+  del(int? id) async {
```

Comment:
Currently, the app uses the BLoC pattern which responsible for this type of operation. I would suggest this to move to LableBloc and so whenever the label is deleted update the getLables stream.

Reply:
can you please guide sir

Reply:
@nightscape would you like to help here? You've been added as reviewer here.

Reply:
@burhanrashid52 though I would love to get my hands dirty with WhatTodo, I'm just in the process of converting a few libraries to Scala 3, so my Open Source time is already booked out...

Reply:
@nigthscape okay no problem. @gourav1486 First you need to move this delete method in their respective bloc. i.e LabelBloc and ProjectBloc and call delete method on the bloc. And then when it successfully deletes it, then call `refresh()` also you need to revert ProjectRow and LabelsRow to StatelessWidget. Can you please add a test for this deletion? You can refer to tests in integration_tests.

Reply:
done but I was unable to make labelRow stateless and its my first open source contribution so I have no idea about  adding test about deletion can you please guide me?

Reply:
You check the existing test code in app that might help to get started with.